lib/sles4sap_publiccloud.pm

```diff
@@ -182,6 +184,70 @@ sub sles4sap_cleanup {
 
 }
 
+=head2 get_job_state
+    Get whether a specified job is still running or not
+=cut
+
+sub get_job_state {
+    my ($job_id) = @_;
+
+    my $url = "https://openqa.suse.de/api/v1/jobs/$job_id";
+
+    my $json_data = script_output("curl -s '$url'");
+
+    my $job_data = eval { decode_json($json_data) };
+    if ($@) {
+        warn "Failed to decode JSON data for job $job_id: $@";
+        return 0;  # Assume job is still running if we can't get its state
+    }
+
+    my $job_state = $job_data->{job}->{state} // '';
+
+    if ($job_state eq 'running') {
+        return 0;  # Job is still running
+    }
+    else {
+        return 1;  # Job has stopped
+    }
+}
+
+
+=head2 get_peerings_azure
+    Get active peerings for Azure jobs
+=cut
+
+sub get_peerings_azure {
```

Comment:
This function can be moved in qesapdeployment.pm with test in https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/t/15_qesap_azure.t

Please name it something like `qesap_az_get_peering_active` 

Please merge functionality (if possible) with `qesap_az_get_peering_name`

Reply:
I can move it, no problem. But I'm not sure about the merge. I mean, qesap_az_get_peering_name is about grabbing the name of the peering for a specific RG, while this one is about returning all active peerings. Are you sure we don't want those 2 separate, so that the names clearly reflect what they're doing?

Reply:
fine not to merge them